extensions/aris/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,493 @@
+		password?: string,
+		rememberPassword?: boolean,
+	}) {
+		this.setPrefilledValues(prefilledValues);
+	}
+
+	public async showDialog(
+		callback: (e: IEndPointsResponse, rememberPassword: boolean) => any,
+		callbackCancel?: () => any,
+		callbackError?: (e: IControllerError) => any,
+	): Promise<void> {
+		this.createDialog();
+		this.toggleSignInButton();
+		this.toggleErrorText();
+		this.callback = callback;
+		this.callbackCancel = callbackCancel;
+		this.callbackError = callbackError;
+		azdata.window.openDialog(this.dialog);
+	}
+
+	private createDialog(): void {
+		this.dialog = azdata.window.createModelViewDialog(this.DialogTitle);
+		this.dialog.registerContent(async view => {
+			this.modelBuilder = view.modelBuilder;
+
+			let tc = new TitledContainer(this.modelBuilder);
+			tc.title = localize('bigDataClusters.controllerCredentialTitle', 'Sign In to Controller');
+			tc.setTitleMargin(6, 0, 11, 0);
+			tc.setPadding(15, 30, 0, 30);
+			tc.contentTopLine = true;
+
+			let allContainer: azdata.FlexContainer = this.modelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'column',
+					alignItems: 'stretch',
+					width: '100%'
+				})
+				.component();
+
+			let urlInputContainer: azdata.FlexContainer = this.modelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			let urlInputText = this.modelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({ value: 'URL' }).component();
+
+			this.urlInputBox = this.modelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({ placeHolder: 'url', value: this.prefilledValues.url }).component();
+			this.urlInputBox.onTextChanged(e => this.toggleSignInButton());
+
+			urlInputContainer.addItem(urlInputText, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'width': '80px'
+				}
+			});
+
+			urlInputContainer.addItem(this.urlInputBox, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%'
+				}
+			});
+
+			allContainer.addItem(urlInputContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'width': '100%'
+				}
+			});
+
+			let usernameInputContainer: azdata.FlexContainer = this.modelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			let usernameInputText = this.modelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({ value: 'Username' }).component();
+
+			this.usernameInputBox = this.modelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({ placeHolder: 'username', value: this.prefilledValues.username }).component();
```

Comment:
Should `admin` be the placeholder? Or do we need one given you have the label?

Reply:
no. because it can be changed.